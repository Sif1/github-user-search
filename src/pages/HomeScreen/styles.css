.screen-wrapper {
    text-align: center;
    background-color: #628283;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .home-title {
      font-size: calc(30px + 2vmin);
      color: white;
  }
  .description-text {
      font-size: calc(20px);
      color: #b3f0ce;
  }

  .button-style {
      margin: 10px;
      padding-left: 6px;
      padding-right: 6px;
      border-radius: 8px;
  }