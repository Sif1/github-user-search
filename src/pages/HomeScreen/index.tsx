import React, { useState } from 'react';
import './styles.css';
import { useNavigate } from "react-router-dom";

function HomeScreen() {

const [searchValue, setSearchValue] = useState("");
const handleChange = (event: any) => {
      setSearchValue(event.target.value);
    };
const navigate = useNavigate();

const handleSearch = () => {
    navigate(`/${searchValue}`);
}

  return (
    <div className="screen-wrapper">
        <h1 className="home-title">
          Git-A-User!
        </h1>
        <p className="description-text">
          Type in a github username to fetch some cool info about a user.
        </p>
        <div>
            <input
            type="text"
            placeholder="Github username"
            value={searchValue}
            onChange={handleChange}

        />
            <button 
            className="button-style"
            disabled={searchValue ? false : true}
            onClick={handleSearch}>
                Search
            </button>
      </div>
    </div>
  );
}

export default HomeScreen;
