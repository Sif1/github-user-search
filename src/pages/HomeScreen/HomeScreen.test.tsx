/* eslint-disable import/first */
const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
   useNavigate: () => mockedUsedNavigate,
 }));

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from './';

test('renders homescreen correctly - snapshot', () => {
  const tree = renderer
    .create(<HomeScreen/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('HomeScreen render title', () => {
    render(<HomeScreen />);
    const titleText = screen.getByText(/Git-A-User!/i);
    expect(titleText).toBeInTheDocument();
  });

test('search button should be disabled on first render', () => {
    render(<HomeScreen />);
    const searchButton = screen.getByRole('button');
    expect(searchButton).toBeDisabled();
  });

  test('search button should be enabled when input has a value', () => {
    render(<HomeScreen />);
    userEvent.type(screen.getByPlaceholderText(/Github username/i), "Sif1");
    const searchButton = screen.getByRole('button');
    expect(searchButton).toBeEnabled();
  });


