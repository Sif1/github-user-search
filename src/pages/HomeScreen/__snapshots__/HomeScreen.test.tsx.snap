// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homescreen correctly - snapshot 1`] = `
<div
  className="screen-wrapper"
>
  <h1
    className="home-title"
  >
    Git-A-User!
  </h1>
  <p
    className="description-text"
  >
    Type in a github username to fetch some cool info about a user.
  </p>
  <div>
    <input
      onChange={[Function]}
      placeholder="Github username"
      type="text"
      value=""
    />
    <button
      className="button-style"
      disabled={true}
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`renders homescreen correctly 1`] = `
<div
  className="screen-wrapper"
>
  <h1
    className="home-title"
  >
    Git-A-User!
  </h1>
  <p
    className="description-text"
  >
    Type in a github username to fetch some cool info about a user.
  </p>
  <div>
    <input
      onChange={[Function]}
      placeholder="Github username"
      type="text"
      value=""
    />
    <button
      className="button-style"
      disabled={true}
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
